docker-compose.yml

```diff
@@ -107,9 +107,6 @@ services:
       - KAFKA_BOOTSTRAP_SERVERS=kafka:29092
       - INVENTORY_LOG_LEVEL=ERROR
       - KAFKA_SECONDARY_TOPIC_ENABLED=True
-    links:
```

Comment:
`depends_on` does not wait for a service to be up, this can lead to hostnames like `db` and `kafka` to be unreachable.
However `links` does need to wait to have the right IP to link to, that is why there are both. 

We could try to remove the depends_on.

Reply:
I did not had problem with the dependent runs, only when it is being shut down the order seemed somewhat random. First it shut down kafka and then the zookeeper was stuck waiting for a connection to kafka.

I'll investigate.

Reply:
The `depends_on` will start the containers in the order. They would have an IP and DNS that can be resolved by the next dependent service. Neither `depends_on` nor `links` wait for the service to be up and running. [Documentation](https://github.com/docker/docker.github.io/blob/master/compose/startup-order.md) suggest to handle connections resiliently in the application, or, some "wait for it" script.

[`links`](https://github.com/docker/docker.github.io/blob/master/compose/compose-file/index.md#links) is a legacy way of connecting containers within their own network. `links` is not needed as the containers are reachable w/o it.

Reply:
`links:` is not a legacy feature, only the `--links` flag[1].`links:` is part of the "user-defined networks" features that are references and recommended to use instead.

When we added the whole stack we added `links:` on purpose[2]. 
It may work properly and fast enough on some systems to get everything up in time, but not on all. 
For example, I still have to separately start inventory services, because they fail to reach kafka, as it isn't ready in time.

"wait for it" scripts would of course prevent any issues, and make sure to start only when all dependencies are reachable as well, but we can't add these for all services, without customising their images.

[1] https://docs.docker.com/compose/compose-file/#links
[2] https://github.com/RedHatInsights/compliance-backend/pull/527#issuecomment-663574839

Reply:
Sure, I'll revert it.

I do understand it that the `links:` in YAML sets the flag `--link` (for each line) when starting a container. It seems that the links (or `--link` param) was kept there to have the possibility to set an alias name. A hostname alias is not configured anywhere.

From the https://docs.docker.com/compose/networking/#links:
> Links allow you to define extra aliases by which a service is reachable from another service. They are not required to enable services to communicate - by default, any service can reach any other service at that service’s name.

But, I understand that you might have a different version of docker/podman where it can work differently.

Reply:
Reverted back to just remove duplicate `links` declarations.